browser/android/background_video/background_video_playback_tab_helper_browsertest.cc

```diff
@@ -0,0 +1,388 @@
+          .ExtractBool());
+  EXPECT_TRUE(
+      content::EvalJs(
+          web_contents(),
+          "window.ytcfg.get(\"WEB_PLAYER_CONTEXT_CONFIGS\").WEB_PLAYER_CONTEXT_"
+          "CONFIG_ID_MWEB_WATCH.serializedExperimentFlags.includes(\"html5_"
+          "picture_in_picture_blocking_onresize=false\")")
+          .ExtractBool());
+  EXPECT_TRUE(
+      content::EvalJs(
+          web_contents(),
+          "window.ytcfg.get(\"WEB_PLAYER_CONTEXT_CONFIGS\").WEB_PLAYER_CONTEXT_"
+          "CONFIG_ID_MWEB_WATCH.serializedExperimentFlags.includes(\"html5_"
+          "picture_in_picture_blocking_document_fullscreen=false\")")
+          .ExtractBool());
+  EXPECT_TRUE(
+      content::EvalJs(
+          web_contents(),
+          "window.ytcfg.get(\"WEB_PLAYER_CONTEXT_CONFIGS\").WEB_PLAYER_CONTEXT_"
+          "CONFIG_ID_MWEB_WATCH.serializedExperimentFlags.includes(\"html5_"
+          "picture_in_picture_blocking_standard_api=false\")")
+          .ExtractBool());
+  EXPECT_TRUE(
+      content::EvalJs(
+          web_contents(),
+          "window.ytcfg.get(\"WEB_PLAYER_CONTEXT_CONFIGS\").WEB_PLAYER_CONTEXT_"
+          "CONFIG_ID_MWEB_WATCH.serializedExperimentFlags.includes(\"html5_"
+          "picture_in_picture_logging_onresize=false\")")
+          .ExtractBool());
+
+  // Verify the other flags were not modified.
+  EXPECT_TRUE(
+      content::EvalJs(
+          web_contents(),
+          "window.ytcfg.get(\"WEB_PLAYER_CONTEXT_CONFIGS\").WEB_PLAYER_CONTEXT_"
+          "CONFIG_ID_MWEB_WATCH.serializedExperimentFlags.includes(\"another_"
+          "flag_for_testing=true\")")
+          .ExtractBool());
+}
+
+IN_PROC_BROWSER_TEST_F(AndroidBackgroundVideoPlaybackBrowserTest,
+                       TestSamePageNavigationInjection) {
+  const GURL url = https_server_.GetURL("youtube.com", "/ytcfg_mock.html");
+
+  content::NavigateToURLBlockUntilNavigationsComplete(web_contents(), url, 1,
+                                                      true);
+
+  ASSERT_TRUE(
+      content::EvalJs(web_contents(), "!!document.getElementById('link1')")
+          .ExtractBool());
+  ASSERT_TRUE(
+      content::EvalJs(web_contents(), "!!document.getElementById('link2')")
+          .ExtractBool());
+
+  // Click the first anchor link with `id=link1` to navigate to "#section1".
+  ClickAndWaitForNavigation(u"document.getElementById('link1').click();");
+
+  // Verify navigation to the same page with the appropriate hash.
+  EXPECT_EQ(web_contents()->GetVisibleURL().ref(), "section1");
+
+  // Click the second anchor link with `id=link2` to navigate to "#section2".
+  ClickAndWaitForNavigation(u"document.getElementById('link2').click();");
+
+  // Verify navigation.
+  EXPECT_EQ(web_contents()->GetVisibleURL().ref(), "section2");
+
+  // Verify the replace method was called exactly 5 times
+  // even if multiple same page navigations.
+  EXPECT_EQ(5, content::EvalJs(web_contents(), kReplaceCallCount).ExtractInt());
```

Comment:
we should check these before and after same page navigation

Reply:
Feedback addressed by commit 👉 [`4294b46` (#28593)](https://github.com/brave/brave-core/pull/28593/commits/4294b46fd66c8bb6ea5032734337b3375bdbec27)